reference/test/matrix/csr_kernels.cpp

```diff
@@ -1760,4 +1764,87 @@ TYPED_TEST(Csr, CanGetSubmatrix2)
 }
 
 
+TYPED_TEST(Csr, CanGetSubmatrixWithIndexSet)
+{
+    using Vec = typename TestFixture::Vec;
+    using Mtx = typename TestFixture::Mtx;
+    using T = typename TestFixture::value_type;
+    using index_type = typename TestFixture::index_type;
+    auto mat = gko::initialize<Mtx>(
+        {
+            I<T>{1.0, 3.0, 4.5, 0.0, 2.0},   // 0
+            I<T>{1.0, 0.0, 4.5, 7.5, 3.0},   // 1
+            I<T>{0.0, 3.0, 4.5, 0.0, 2.0},   // 2
+            I<T>{0.0, -1.0, 2.5, 0.0, 2.0},  // 3
+            I<T>{1.0, 0.0, -1.0, 3.5, 1.0},  // 4
+            I<T>{0.0, 1.0, 0.0, 0.0, 2.0},   // 5
+            I<T>{0.0, 3.0, 0.0, 7.5, 1.0}    // 6
+        },
+        this->exec);
+
+    ASSERT_EQ(mat->get_num_stored_elements(), 23);
+
+    {
```

Comment:
Test for empty IndexSet and full IndexSet maybe?

Reply:
Also perhaps with only one empty set?